loopy/transform/buffer.py

```diff
@@ -248,26 +240,6 @@ def buffer_array_for_single_kernel(kernel, callables_table, var_name,
 
     # }}}
 
-    # {{{ caching
-
-    from loopy import CACHING_ENABLED
-
-    cache_key = (kernel, var_name,
-            tuple(buffer_inames),
-            PymbolicExpressionHashWrapper(init_expression),
-            PymbolicExpressionHashWrapper(store_expression), within,
```

Comment:
How come special treatment of pymbolic expressions in arguments is no longer needed now?

Reply:
It doesn't break tests and I don't understand what [these]( https://github.com/inducer/loopy/blob/9e69cf4b2dedad95e5f52ac5d7a8fbe1a08a8c80/loopy/tools.py#L131-L143) were supposed to do.

Reply:
I suspect that there just isn't a test that passes a pymbolic expression to this argument. Maybe we should add one. `LoopyKeyBuilder` won't recognize pymoblic expressions (i.e. dispatch to `update_for_pymbolic_expression`) on its own. The wrapper helps with that. (Look at the dispatch mechanism in `KeyBuilder` in pytools.)

Reply:
Preprocessed all the args to  convert pym.Expression -> PymbolicExpressionHashWrapper. Also added a test that's able to memoize functions with pymbolic expressions.